tests/Unit/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Test_WrapGh.cpp

```diff
@@ -0,0 +1,286 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <algorithm>
+#include <array>
+#include <limits>
+#include <string>
+
+#include "DataStructures/DataBox/Prefixes.hpp"  // IWYU pragma: keep
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Minkowski.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/WrapGh.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+// IWYU pragma: no_forward_declare Tags::deriv
+
+namespace {
+template <typename SolutionType>
+void compare_different_wrapped_solutions(const SolutionType& solution_1,
+                                         const SolutionType& solution_2) {
+  CHECK_FALSE(solution_1 == solution_2);
+  CHECK(solution_1 != solution_2);
+}
+
+template <typename SolutionType>
+void test_generalized_harmonic_solution(
+    const SolutionType& solution,
```

Comment:
Why not make this a variadic function as follows:
```cpp
template <typename SolutionType>
void test_generalized_harmonic_solution(const Args&... args) noexcept {
  SolutionType solution{args...};
  GeneralizedHarmonic::Solutions::WrapGh<SolutionType> wrapped_solution{args...};
  // Other testing code.

  // Test serialization:
  test_serialization(wrapped_solution);

  test_copy_and_move(wrapped_solution);
}
```

Then the test case is just:
```cpp
SPECTRE_TEST_CASE("Unit.PointwiseFunctions.AnalyticSolutions.Gr.WrapGh",
                  "[PointwiseFunctions][Unit]") {
  test_generalized_harmonic_solution<gr::Solutions::Minkowski<1>>();
  test_generalized_harmonic_solution<gr::Solutions::Minkowski<2>>();
  test_generalized_harmonic_solution<gr::Solutions::Minkowski<3>>();

  const double mass = 0.5;
  const std::array<double, 3> spin{{0.1, 0.2, 0.3}};
  const std::array<double, 3> center{{1.0, 3.0, 2.0}};
  test_generalized_harmonic_solution<gr::Solutions::KerrSchild>(mass, spin, center);
  
  const double mass2 = 0.4;
  const std::array<double, 3> spin2{{0.4, 0.3, 0.2}};
  const std::array<double, 3> center2{{4.0, 1.0, 3.0}};
  test_generalized_harmonic_solution<gr::Solutions::KerrSchild>(mass2, spin2, center2);

  test_construct_from_options();
}
```